addons/pfconfig/comparator/dumper.pl

```diff
@@ -0,0 +1,143 @@
+#!/usr/bin/perl
+
+=head1 NAME
+
+dumper.pl
+
+=head1 SYNOPSIS
+
+dumper.pl <codebase>
+
+=head1 DESCRIPTION
+
+Dumps the configuration for a codebase
+
+No need to use it directly, use addons/pfconfig/comparator/config-comparator.sh
+
+=cut
+
+use strict;
+use warnings;
+
+use Sereal::Encoder;
+use Data::Dumper;
+
+my $BASE = $ARGV[1];
+use lib $ARGV[0];
+
+my $ENCODER = Sereal::Encoder->new;
+our %configs;
+
+sub {
+  use pf::config;
+  my @exported = @pf::config::EXPORT;
+  my @badvalues = ('%ConfigProvisioning');
+  @exported = grep { !($_ ~~ @badvalues ) } @exported;
+  dump_module('pf::config', @exported);
+}->();
```

Comment:
I just want to know why.
That is what I thought the variable scoping rules of javascript is different than most languages.
That is why they use anonymous functions in javascript.

It would be easier to use blocks
{ 
  my @exported = @pf::config::EXPORT;
}

{
   my @exported = @pf::somemoudle::EXPORT;
}

Since it is a test script no need to waste the effort to do so.